src/docfx/lib/markdown/validation/DocsValidationExtension.cs

```diff
@@ -80,6 +82,32 @@ internal static class DocsValidationExtension
                         }
                     }
 
+                    CodeBlockItem? codeBlockItem = null;
+                    if (node.GetType().Name.Equals(nameof(CodeBlock)) && context.TripleColonCount == 0)
+                    {
+                        codeBlockItem = new CodeBlockItem
+                        {
+                            Type = CodeBlockTypeEnum.CodeBlock,
+                            SourceInfo = node.GetSourceInfo(),
+                        };
+                    }
+                    else if (node is FencedCodeBlock fencedCodeBlock)
+                    {
+                        codeBlockItem = new CodeBlockItem
+                        {
+                            Type = CodeBlockTypeEnum.FencedCodeBlock,
+                            Info = fencedCodeBlock.Info,
+                            Arguments = fencedCodeBlock.Arguments,
+                            IsOpen = fencedCodeBlock.IsOpen,
+                            SourceInfo = fencedCodeBlock.GetSourceInfo(),
+                        };
+                    }
+
+                    if (codeBlockItem != null)
+                    {
+                        codeBlockItemList.Add(codeBlockItem);
+                    }
```

Comment:
extract a method for code block validation

Reply:
ok